src/comm.cc

```diff
@@ -424,7 +424,7 @@ comm_init_opened(const Comm::ConnectionPointer &conn,
     debugs(5, 5, HERE << conn << " is a new socket");
 
     assert(!isOpen(conn->fd)); // NP: global isOpen checks the fde entry for openness not the Comm::Connection
-    fd_open(conn->fd, FD_SOCKET, note);
+    fd_open(conn->fd, FD_SOCKET, SBuf(note));
```

Comment:
Since this PR is about fixing the compile error I don't want to complicate it any further. But do want to hear these ideas. Can you post them to squid-dev so we can go over it separately please.

FYI, This is not much of a performance impact at all. The data-copy part cancels out with the old code removed. The addition is the "allocate" part - which for SBuf will almost always be from the short-strings pool. So just a matter of a few pointer operations instead of a regular memory alloc() syscall it would have been.

Reply:
> do want to hear these ideas. Can you post them to squid-dev so we can go over it separately

Done at http://lists.squid-cache.org/pipermail/squid-dev/2019-February/009503.html

> FYI, This is not much of a performance impact at all

FYI, "death by a thousand cuts" is one of the reasons "why we can't have nice things".